events/tests/importers/test_espoo.py

```diff
@@ -0,0 +1,449 @@
+import io
+from http.client import HTTPMessage, HTTPResponse
+from unittest.mock import Mock, patch
+
+import factory
+import pytest
+import pytz
+from django.conf import settings
+from django_orghierarchy.models import Organization
+from faker import Faker
+
+from events.importer.espoo import (
+    _build_pre_map_to_id,
+    _get_data,
+    _get_origin_objs,
+    _import_origin_objs,
+    _list_data,
+    _post_map_to_self_id,
+    _split_common_objs,
+    EspooImporter,
+    EspooImporterError,
+    parse_jsonld_id,
+)
+from events.models import Event, Keyword, Place
+from events.tests.factories import (
+    DataSourceFactory,
+    KeywordFactory,
+    OrganizationFactory,
+    PlaceFactory,
+)
+
+
+class BaseDataFactory(factory.DictFactory):
+    id = factory.Sequence(lambda n: f"origin:{n}")
+    data_source = "origin_ds"
+
+    class Meta:
+        abstract = True
+
+
+class EventDataFactory(BaseDataFactory):
+    super_event = None
+    super_event_type = None
+    publisher = None
+
+
+def test_parse_jsonld_id():
+    assert ["tprek:8100", "espoo:asdfasdf"] == (
+        [
+            parse_jsonld_id(
+                {
+                    "@id": "https://api.espoo.localhost/linkedevents/v1/place/tprek:8100/",
+                }
+            ),
+            parse_jsonld_id(
+                {
+                    "@id": "https://api.espoo.localhost/linkedevents/v1/place/espoo:asdfasdf/",
+                }
+            ),
+        ]
+    )
+
+
+@pytest.fixture
+def sleep(monkeypatch):
+    class Sleep:
+        call_count = 0
+
+        def sleep(self, t):
+            self.call_count += 1
+
+    sleep_instance = Sleep()
+
+    monkeypatch.setattr("time.sleep", sleep_instance.sleep)
+    return sleep_instance
```

Comment:
Could use `Mock` instead:
```suggestion
@pytest.fixture
def sleep(monkeypatch):
    mock_sleep = Mock()  # or a MagicMock
    monkeypatch.setattr("time.sleep", mock_sleep)
    return mock_sleep
```

Or if you don't like mixing & matching:
```suggestion
@pytest.fixture
def sleep():
    with patch("time.sleep") as mock:  # not completely sure on the target; maybe "events.importer.sleep" if this one doesn't work?
        yield mock
```